op-node/rollup/derive/engine_update.go

```diff
@@ -157,6 +178,10 @@ func confirmPayload(ctx context.Context, log log.Logger, eng ExecEngine, fc eth.
 			return nil, BlockInsertTemporaryErr, fmt.Errorf("failed to make the new L2 block canonical via forkchoice: %w", err)
 		}
 	}
+	// at this point the block is either inserted, or not valid, so clear it from the async gossiper
```

Comment:
I'm not sure this assertion is correct.  In the code above if we get status codes from either the `eng.NewPayload` or `eng.ForkchoiceUpdated` calls we will return but won't have cleared the cached payload which I think means we'll try to insert it again on the next time round the loop. I think anywhere we're returning `BlockInsertPayloadErr` we should clear the async gossip as well.

Reply:
Good call. I have added calls to Clear prior to any return besides Temporary Errors. I'll consult with @trianglesphere to make sure these are lined up correctly.